### ISSUE TITLE
youtube-dl does not download viki subtitles

### ISSUE THREAD

**Issue by fursiol0800** [ytdl-org/youtube-dl#29224]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.06.06**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
youtube-dl "https://www.viki.com/videos/1178694v" --write-sub --sub-lang en --sub-format vtt --
skip-download --verbose
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'https://www.viki.com/videos/1178694v', u'--write-sub', u'--sub-lang', u'en', u'--sub-format', u'vtt', u'--skip-download', u'--verbose']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.06.06
[debug] Python version 2.7.16 (CPython) - Linux-5.10.17-v8+-aarch64-with-debian-10.9
[debug] exe versions: ffmpeg 4.1.6-1, ffprobe 4.1.6-1
[debug] Proxy map: {}
[viki] 1178694v: Downloading video JSON
[viki] 1178694v: Downloading MPD manifest
WARNING: [viki] Unknown MIME type image/jpeg in DASH manifest
[debug] Default format spec: bestvideo+bestaudio/best
[info] Writing video subtitles to: Love of Thousand Years - Episode 9-1178694v.en.vtt
WARNING: Unable to download subtitle for "en": Unable to download webpage: HTTP Error 400: Bad Request (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

## Description

I tried to download viki subtitles of many viki videos but recently youtube-dl doesn't download any subtitles in any language.

**Comment by kurumushi** [comment]
Having the same problem, here is some additional information:

My test video: https://www.viki.com/videos/1179709v-you-are-my-hero-episode-1 (since it's available in my region)

The URL used by yt-dl to grab the subtitle:
https://api.viki.io/v4/videos/1179709v/subtitles/en.vtt?app=100005a&t=1622942642&site=www.viki.com&sig=9ea3b47a68f7aef646ba9ddf629ddbbc670451b7
json returned by the server says: "invalid signature" 

The URL listed in the .json file for the video when viewing on the viki website:
https://api.viki.io/v4/videos/1179709v/subtitles/en.vtt?app=100000a&t=1622942181&sig=6ad37ce14a1d10a59ff8b85cc8810148f8885532

Hopefully this helps point someone in the right direction.

**Comment by dirkf** [comment]
Fixed in #30596.
